internal/executor/job/submit.go

```diff
@@ -147,6 +153,29 @@ func (submitService *SubmitService) submitPod(job *SubmitJob) (*v1.Pod, error) {
 	return pod, err
 }
 
+// Sanitizes pod resources that may be used during scheduling but are invalid at runtime.
+func (submitService *SubmitService) sanitizePodResources(pod *v1.Pod) {
```

Comment:
This function can be simplified to a simple `sanitizePodResources` function, no need for it to be a receiver method on SubmitService, same for `sanitizeResourceList`